cfn-resources/online-archive/cmd/resource/resource.go

```diff
@@ -272,17 +178,15 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return validateProgress(ctx, client, currentModel, "PENDING")
 	}
 
-	a, _ := ArchiveExists(context.Background(), client, currentModel)
-	if a.State == "DELETED" {
+	if a, _ := ArchiveExists(ctx, client, currentModel); *a.State == "DELETED" {
```

Comment:
```suggestion
        deleted := "DELETED" 
	if a, _ := ArchiveExists(ctx, client, currentModel); util.AreStringPtrEqual(a.State, &deleted) {
```
Small suggestions for avoiding nil dereferencing. We can create a variant of this method to receive a string value as well.

Reply:
I've created ArchiveDeleted that I think that it solves this better